README.md

```diff
@@ -1,152 +1,197 @@
+  id: 'helloworld',
+  version: '1.0',
+  name: 'Hello World Workflow',
+  description: 'Inject Hello World',
+  start: 'Hello State',
+  states: [
+    {
+      type: 'inject',
+      name: 'Hello State',
+      end: true,
+      data: {
+        result: "Hello World!"
+      }
+    } as Specification.Injectstate
+  ]
+};
+```
+
+
+#### Load a file JSON/YAML to a Workflow instance
+
+```typescript
+import { Specification, WorkflowConverter } from '@severlessworkflow/sdk-typescript';
+
+const workflow: Specification.Workflow = WorkflowConverter.fromString(source);
+```
+Where `source` is a JSON or a YAML string.
+
+
+#### Parse a Workflow instance to JSON/YAML
+
+Having the following workflow instance:
+
+```typescript
+import { workflowBuilder, injectstateBuilder, Specification } from '@severlessworkflow/sdk-typescript';
+
+const workflow: Specification.Workflow = workflowBuilder()
+  .id("helloworld")
+  .version("1.0")
+  .name("Hello World Workflow")
+  .description("Inject Hello World")
+  .start("Hello State")
+  .states([
+    injectstateBuilder()
+      .name("Hello State")
+      .data({
+        "result": "Hello World!"
+      })
+      .end(true)
+      .build()
+  ])
+  .build();
+```
+
+You can convert it to its string representation in JSON or YAML format 
+by using the static methods `toJson` or `toYaml` respectively:
+
+```typescript
+import { WorkflowConverter } from '@severlessworkflow/sdk-typescript';
+
+const workflowAsJson: string = WorkflowConverter.toJson(workflow);
+```
+
+```typescript
+import { WorkflowConverter } from '@severlessworkflow/sdk-typescript';
+
+const workflowAsYaml: string = WorkflowConverter.toYaml(workflow);
+```
+
+
+#### Validate workflow definitions
+
+The sdk provides a way to validate if a workflow object is compliant with the serverlessworkflow specification.
+
+`WorkflowValidator` class provides a validation method: 
+
+- `validate(): boolean`
+
+```typescript
+import { WorkflowValidator, Specification } from '@severlessworkflow/sdk-typescript';
+
+const workflow: Specification.Workflow = {
+  id: 'helloworld',
+  version: '1.0',
+  name: 'Hello World Workflow',
+  description: 'Inject Hello World',
+  start: 'Hello State',
+  states: [
+    {
+      type: 'inject',
+      name: 'Hello State',
+      end: true,
+      data: {
+        result: "Hello World!"
+      }
+    } as Specification.Injectstate
+  ]
+};
+const workflowValidator: WorkflowValidator = new WorkflowValidator(workflow);
+if (!workflowValidator.validate()) {
+  workflowValidator.errors.forEach(error => console.error(error.message));
+}
+```
+
+You can also validate parts of a workflow using `validators`:
+
+```typescript
+import { ValidateFunction } from 'ajv';
+import { validators, Specification } from '@severlessworkflow/sdk-typescript';
+
+const injectionState: Specification.Injectstate = workflow.states[0];
+const injectionStateValidator: ValidateFunction<Specification.Injectstate> = validators.get('Injectstate');

```

Comment:
`+` if I where a user of the sdk, I don't want to learn the keys for each element to retrieve the validator from `validators`

Reply:
The 'keys' are the type names of Specification but I agree with you, it could be improved. I need to double check but I don't think we can use the type directly in TS like `validator.get(Specification.Injectstate)` because the type doens't exist at runtime. Anyhow, good idea, there is room for improvement here.

One side note though, more regarding the schema, I think Events/Retries/Functions cannot be used by themselves because they are based on custom properties that don't exist in the JSON Schema specification. So they can be "pointed at" but not used alone. I might be wrong though. (See little chat in the [JSON Schema Slack](https://app.slack.com/client/T5CHC2TAN/C5CF75URH/thread/C5CF75URH-1619796577.402700))